test/AutoDiff/Parse/transpose_attr_parse.swift

```diff
@@ -0,0 +1,92 @@
+// RUN: %target-swift-frontend -parse -verify %s
+
+/// Good
+
+@transpose(of: foo)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo(_:_:))
+func transpose(v: Float) -> Float
+
+@transpose(of: wrt, wrt: 0)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo, wrt: 0)
+func transpose(v: Float) -> Float
+
+@transpose(of: foo, wrt: (0, 1))
+func transpose(v: Float) -> (Float, Float)
+
+@transpose(of: foo, wrt: (self, 0, 1, 2))
+func transpose(v: Float) -> (Float, Float, Float, Float)
+
+// Qualified declaration.
+@transpose(of: A.B.C.foo(x:y:_:z:))
+func transpose(v: Float) -> Float
+
+// Qualified operator.
+// TODO(TF-1065): Consider disallowing qualified operators.
+@transpose(of: Swift.Float.+)
+func transpose(v: Float) -> Float
+
+// Qualified leading-period operator (confusing).
+// TODO(TF-1065): Consider disallowing qualified operators.
+@transpose(of: Swift.Float..<)
+func transpose(v: Float) -> Float
+
+// `init` keyword edge case.
+@transpose(of: Swift.Float.init(_:))
+func transpose(v: Float) -> Float
+
+// `subscript` keyword edge case.
+@transpose(of: Swift.Array.subscript(_:))
```

Comment:
Could you add test cases for specialized generic types? Like: `@transpose(of: Foo<Int>.Bar.funcname)`

Reply:
Specialized generic type test added in https://github.com/apple/swift/pull/27545/commits/c736938911f0d2c4b5fd08968c365bc58e1c0539. It passes, as expected.